src/modules.rs

```diff
@@ -249,6 +252,14 @@ impl<'ast, 'sess, 'c> ModResolver<'ast, 'sess> {
             // mod foo;
             // Look for an extern file.
             self.find_external_module(item.ident, &item.attrs, sub_mod)
+                .or_else(|err| match err.kind {
+                    ModuleResolutionErrorKind::NotFound { file: _ }
+                        if self.ignore_missing_submod =>
+                    {
+                        Ok(None)
+                    }
+                    _ => Err(err),
+                })
```

Comment:
I'd like to rework this with `Error`, `Warn`, and `Ignore` variants in mind. I think the only difference is that if `Warn` is set we'll emit an error message, but still continue with `Ok(None)`.

Reply:
Thank you for the review! When emitting a message for a 'warn', is it sufficient to simply use `eprintln!` to output to stderr? Or is there a more appropriate method you could suggest?

Reply:
Fixed. https://github.com/rust-lang/rustfmt/pull/5611/commits/f678369ce39f5d2591bc94a8ef1e81bf45053269

For now, I implemented direct output with eprintln.
(However, I think I might be better off using FormatReport.)